Add BasedPyright section and Integrations section in README (#634)
KotlinIsland/basedmypy: README.md

See the [features](#features) for more information, or [the docs](https://kotlinisland.github.io/basedmypy/based_features.html) for a comprehensive list.

### BasedPyright

> Also, take a look at [BasedPyright](https://github.com/DetachHead/basedpyright), a based type checker in the flavour of Pyright!

Reviewer: "flavour of Pyright" sounds kinda weird

Author: How would you wright it?

Reviewer: maybe just
```suggestion
Also, take a look at [BasedPyright](https://github.com/DetachHead/basedpyright), a based type checker based on Pyright!
```
